spring-cloud-gcp-samples/spring-cloud-gcp-integration-storage-sample/src/main/java/com/example/GcsSpringIntegrationApplication.java

```diff
@@ -129,6 +128,9 @@ public MessageHandler outboundChannelAdapter(Storage gcs) {
 		GcsMessageHandler outboundChannelAdapter = new GcsMessageHandler(new GcsSessionFactory(gcs));
 		outboundChannelAdapter.setRemoteDirectoryExpression(
 				new ValueExpression<>(this.gcsWriteBucket));
+		outboundChannelAdapter
+				.setFileNameGenerator(message -> message.getHeaders().get(FileHeaders.REMOTE_FILE, String.class) + "."
+						+ System.currentTimeMillis());
```

Comment:
Is it really what we want to add in the end of file name as an extension?

Reply:
Seemed like a way to avoid filename collision every time the sample is run. I can remove it.

Reply:
Well, you may just document that default file writing mode is `FileExistsMode.REPLACE`.
So, a new file with the same name will override existing one in the `gcsWriteBucket`.
Or you can configure it to the `FAIL` to let end-user to know that file is already there and has to be removed before the next sample cycle.